digiroad2-api-oth/src/test/scala/fi/liikennevirasto/digiroad2/dataexport/AssetReportCsvExporterSpec.scala

```diff
@@ -0,0 +1,160 @@
+    override val userProvider: UserProvider = mockUserProvider
+  }
+
+  val LupaMunicipalityInfo = MunicipalityInfo(408, 3, "Lapua")
+  val SavitaipaleMunicipalityInfo = MunicipalityInfo(739, 8, "Savitaipale")
+
+  val vvhRoadlinks1 = Seq(VVHRoadlink(388562360L, 408, Seq(Point(0, 0), Point(10, 0)), Municipality, TrafficDirection.UnknownDirection, FeatureClass.AllOthers))
+  val vvhRoadlinks2 = Seq(VVHRoadlink(38856690L, 739, Seq(Point(10, 0), Point(65, 0)), Municipality, TrafficDirection.UnknownDirection, FeatureClass.AllOthers))
+
+  val operatorUser = User(26867454, "k647320", Configuration(None, None, None, None, None, Set(), Set(), Set("operator"), Some("2018-11-27"), None), None)
+
+  val users = List( operatorUser,
+                  User(642513, "l485693", Configuration(None, None, None, None, None, Set(408, 739), Set(), Set(), None, None), None),
+                  User(123456, "lx775599", Configuration(None, None, None, None, None, Set(), Set(), Set("elyMaintainer"), Some("2019-12-07"), None), None) )
+
+
+  test("test function exportAssetsByMunicipality") {
+    OracleDatabase.withDynTransaction {
+      val municipalitiesParam = "408, 739"
+      val assetTypesParam = "130"
+      val fileName = "test_export.csv"
+
+      val municipalitiesList = List(LupaMunicipalityInfo, SavitaipaleMunicipalityInfo)
+
+      val assetReportsList1 = List(AssetReport(130, "Liikennemerkki", "linear", "k647320", DateTime.now().minusDays(2)))
+      val assetReportsList2 = List(AssetReport(130, "Liikennemerkki", "linear", "l485693", DateTime.now().minusDays(5)))
+
+      when(mockUserProvider.getUsers()).thenReturn(users)
+      when(mockMunicipalityDao.getMunicipalitiesNameAndIdByCode(any[Set[Int]])).thenReturn(municipalitiesList)
+
+      when(mockRoadLinkService.getVVHRoadLinksF(408)).thenReturn(vvhRoadlinks1)
+      when(mockRoadLinkService.getVVHRoadLinksF(739)).thenReturn(vvhRoadlinks2)
+
+      when(mockAssetReporterDAO.linearAssetQuery(Seq(388562360L), Seq(130))).thenReturn(assetReportsList1)
+      when(mockAssetReporterDAO.linearAssetQuery(Seq(38856690L), Seq(130))).thenReturn(assetReportsList2)
+
+
+      val municipalities = assetReportCsvExporter.decodeMunicipalitiesToProcess(municipalitiesParam.split(",").map(_.trim.toInt).toList)
+      val assetTypes = assetReportCsvExporter.decodeAssetsToProcess(assetTypesParam.split(",").map(_.trim.toInt).toList)
+
+      val logId = csvDataExporter.insertData(operatorUser.username, fileName, assetTypes.mkString(","), municipalities.mkString(","), false)
+      val generatedDataId = assetReportCsvExporter.exportAssetsByMunicipalityCSVGenerator(assetTypes, municipalities, logId, false)
+
+      val result = csvDataExporter.getById(generatedDataId, false)
+      val finalContent = result match{
+        case Some(x) => x.content match {
+          case Some(content) => content
+          case _ => ""
+        }
+        case _ => ""
+      }
```

Comment:
For me, in a test, this is useless. You are testing tha it should contains something in there so no need to alter it to a empty string because of options. If is None for some reason the test will fail because of a exception and that's fine because it should have a Some().
 
var finalContent = result.get.content.get

Reply:
what is preferable? A failed test due a should condition that is false or a test fail due a None.get exception?